src/main/java/it/gov/pagopa/afm/calculator/repository/CosmosRepository.java

```diff
@@ -201,4 +208,38 @@ private static boolean belongsCI(ValidBundle bundle) {
         && bundle.getCiBundleList() != null
         && !bundle.getCiBundleList().isEmpty();
   }
+
+  /**
+   * @param iterator an iterator of PspSearchCriteria objects to generate filter criteria for the
+   *     psp
+   * @return Criteria an AND/OR concatenation of the global psp filter criteria
+   */
+  private Criteria getPspFilterCriteria(Iterator<PspSearchCriteria> iterator) {
+    PspSearchCriteria pspSearch = iterator.next();
+    var idPspFilter = isEqual("idPsp", pspSearch.getIdPsp());
+    var idChannelFilter =
+        StringUtils.isNotEmpty(pspSearch.getIdChannel())
+            ? isEqual("idChannel", pspSearch.getIdChannel())
+            : idPspFilter;
+    var idBrokerPspFilter =
+        StringUtils.isNotEmpty(pspSearch.getIdBrokerPsp())
+            ? isEqual("idBrokerPsp", pspSearch.getIdBrokerPsp())
+            : idPspFilter;
+    Criteria pspFilterCriteria = and(idPspFilter, and(idChannelFilter, idBrokerPspFilter));
```

Comment:
I think it is not clear. Add `idChannelFilter` and `idBrokerPspFilter` only if they are needed.